status-whisper-usage-spec.md

```diff
@@ -20,289 +36,145 @@ transport layer on top of Whisper.
+partitionTopic := "contact-discovery-" + strconv.FormatInt(partition.Int64(), 10)
 
-Sending a message is fairly easy and relies on the Whisper RPC API, however, some preparation is needed:
+var hash []byte = keccak256(partitionTopic)
+var topicLen int = 4
 
-1. Obtain a public key of the recipient of the message
-2. Add your private key to Whisper using [`shh_addPrivateKey`](https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API#shh_addprivatekey) and save the result as `sigKeyID`
-3. Call [`shh_post`(https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API#shh_post) with the following settings:
-   1. `pubKey` MUST be a hex-encoded public key of the message recipient
-   2. `sig` MUST be set to `sigKeyID`
-   3. `ttl` MUST be at least `10` (it is in seconds)
-   4. `topic` MUST be set accordingly to [Topic](#topic) section and hex-encoded
-   5. `payload` MUST be a hex-encoded string
-   6. `powTime` MAY be arbitrary but should be enough to perform proof-of-work
-   7. `powTarget` MUST be equal or lower than `0.002`.
+if len(hash) < topicLen {
+    topicLen = len(hash)
+}
 
-Note: these instructions are for the Whisper V6 RPC API. If you use Whisper service directly or Go `shhclient`, the parameters might have different types.
+var topic [4]byte
+for i = 0; i < topicLen; i++ {
+    topic[i] = hash[i]
+}
+```
 
-Learn more following [Whisper V6 RPC API](https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API).
+If partitioned topic support is enabled by the Status client, it MUST listen to its paritioned topic. It MUST be generated using the algorithm above and active public key.
 
+### Public chats
 
-### Sending using PFS
+A public chat MUST use a topic derived from a public chat name following the algorithm below:
+```golang
+var hash []byte
+hash = keccak256(name)
 
-When one decides to use PFS, the flow is the same but the payload MUST be additionally encrypted following the [PFS specification](#pfs) before being hex-encoded and passed to `shh_post`.
+topicLen = 4
+if len(hash) < topicLen {
+    topicLen = len(hash)
+}
 
-## Receiving
+var topic [4]byte
+for i = 0; i < topicLen; i++ {
+    topic[i] = hash[i]
+}
+```
 
-Receiving private messages depends on Whisper filters idea. Upon receiving, messages are first matched by a topic and then trying to be decrypted using user's private key.
+### Personal discovery topic
 
-1. Add your private key to Whisper using [`shh_addPrivateKey`](https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API#shh_addprivatekey) and save the result as `sigKeyID`
-2. Call [`shh_subscribe`](https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API#shh_subscribe) with criteria:
-   1. `minPow` MUST be at least `0.002`
-   2. `topics` MUST be list of hex-encoded topics you expect messages to receive from (follow [Topic](#topic) section)
-   3. `allowP2P` MUST be set to `true` if offline messages are supported, otherwise can be `false`.
+Personal discovery topic is used to ???
```

Comment:
@cammellos and I need some help with this one as well.

Reply:
This one is currently not used, we decided to add it as a redundant channel in case we need to migrate topics again

Reply:
Should we get rid of this fragment completely? Also, `in case we need to migrate topics again` -- is this about split discovery topic?

Reply:
I'm commenting it out, keeping spec to V1, but leaving for comment in case we need later